### PR TITLE
Adding Document.analyze_entities() in language package

### PR DESCRIPTION
**Pull request by dhermes**
Also adding system tests and the class needed for the return value.

~~**NOTE**: Has #2164 as diffbase.~~

### CODE REVIEW

File: docs/language-usage.rst [hunk 1]
```diff
@@ -179,17 +179,17 @@ metadata and other properties.
...
-     salience: 0.20193423
+     salience: 0.19960518
      ====================
          name: The Calling of Saint Matthew
-         type: WORK_OF_ART
-     metadata: {'wikipedia_url': 'http://en.wikipedia.org/wiki/index.html?curid=2838808'}
-     salience: 0.03863598
+         type: EVENT
+     metadata: {'wikipedia_url': 'http://en.wikipedia.org/wiki/The_Calling_of_St_Matthew_(Caravaggio)'}
+     salience: 0.038798928
```
Comment (tseaver) [inline docs/language-usage.rst]: Where did these changes come from? And I don't understand the reclassification of Caravaggio's painting from `WORK_OF_ART` to `EVENT` (maybe an *additional* entity for the event the painting depicts, but not a replacement).

Comment (tseaver) [inline docs/language-usage.rst]: If the back-end is going to change what it returns for the same text like this, we are building in more flakiness. :(

**Comment by tseaver** [comment]
1e5c6ddf6335cbf90f186073339fb641f49eb0c4 SGTM.  Do we want to unpack the `metadata` at all, e.g. to get the `wikipedia_url` if present?

7dbdefb99124d7e92d4b80901d65ef990b4ac120 SGTM.

a34109d8eaf7585c7d42a5f63523de163a9a4f89 SGTM.

7730614: See my question / comment.

File: system_tests/language.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            'wikipedia_url': 'http://en.wikipedia.org/wiki/Italy',
+        })
+        # Verify entity 3.
+        self.assertEqual(entity3.name, 'The Calling of Saint Matthew')
+        self.assertEqual(entity3.entity_type, EntityType.EVENT)
+        self.assertTrue(0 < entity3.salience < 0.1)
+        self.assertEqual(entity3.mentions, [entity3.name])
+        wiki_url = ('http://en.wikipedia.org/wiki/'
+                    'The_Calling_of_St_Matthew_(Caravaggio)')
+        self.assertEqual(entity3.metadata, {'wikipedia_url': wiki_url})
```
Comment (daspecster) [inline system_tests/language.py]: Should there be a system test for `document_from_url()` to check Google Storage?

File: docs/language-usage.rst (hunk 1 above)
Comment (dhermes) [inline docs/language-usage.rst]: Ha I know. I'm not sure where I got the original values. I may have copy-pasted but I'm pretty sure I got them from using the [APIs explorer][1]

[1]: https://developers.google.com/apis-explorer/#p/language/v1beta1/language.documents.analyzeEntities

File: system_tests/language.py (hunk 2 above)
Comment (dhermes) [inline system_tests/language.py]: :+1: Good call! I'll do it in another PR?

Comment (daspecster) [inline system_tests/language.py]: SGTM, I don't know if `document_from_blob` is as important?

Comment (dhermes) [inline system_tests/language.py]: They are just sugar. The thing that actually matters is the API request

**Comment by dhermes** [comment]
@tseaver PTAL after I rebased on top of merged #2164.

> Do we want to unpack the metadata at all, e.g. to get the wikipedia_url if present?

I considered it. The spec just says `metadata` is an arbitrary `string->string` map, but it also says 

> Currently, only Wikipedia URLs are provided, if available. The associated key is "wikipedia_url".

WDYT?

----

> If the back-end is going to change what it returns for the same text like this, we are building in more flakiness. :(

I agree, but am not sure what do do about it. (And the change in value might just be docs that are out of date.)

**Comment by tseaver** [comment]
I'd say surface the `wikipedia_url`, if it is present, and leave `metadata` there to catch anything else (maybe they'll add stuff like ticker symbols later?).

If the original values came from docs, and not from the back-end as you tested it, then we can just go with them, and be prepared for flakiness to ensue.

**Comment by dhermes** [comment]
@tseaver PTAL at the latest commit.

File: gcloud/language/test_document.py [hunk 3]
```diff
@@ -102,20 +102,16 @@ def test_analyze_entities(self):
...
+        wiki2 = 'http://en.wikipedia.org/wiki/United_States'
         salience1 = 0.91391456
         salience2 = 0.086085409
         response = {
             'entities': [
                 {
                     'name': name1,
                     'type': EntityType.OTHER,
-                    'metadata': metadata1,
+                    'metadata': {'wikipedia_url': wiki1},
```
Comment (tseaver) [inline gcloud/language/test_document.py]: Maybe leave out the URL for one entry, to test the default is None?

Comment (tseaver) [inline gcloud/language/test_document.py]: Given today's Travis queue, maybe not. :(

**Comment by tseaver** [comment]
LGTM

File: gcloud/language/test_document.py (hunk 3 above)
Comment (dhermes) [inline gcloud/language/test_document.py]: Good call. Needs to be tested. Doing it anyway.

**Pull request merged**